### DISCUSSION TITLE
EarlyStopping every two epochs

### DISCUSSION THREAD

**Discussion by quentinblampey** [Lightning-AI/pytorch-lightning#16231] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello,

My model is training on two different dataloaders: `loader1` is used on even epoch number, and `loader2` for odd numbers.

I'm using EarlyStopping, but I want it to focus only on the even number of epochs, i.e. when we are using`loader1` (thus, I don't want to consider metrics logged from `loader2`).
Do you have any idea how to do this?

I tried to log the loss only for `loader1`, but it says:
```
You called `self.log(loss, ...)` twice in `training_step` with different arguments. This is not allowed
```
I could also multiply the patience by 2, but the loss is usually lower on the `loader2`, so the current lower loss value is updated everything when training on the second loader.
